lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_action.rb

```diff
@@ -401,37 +405,64 @@ def self.test_release(alpha_client, release, test_devices, username = nil, passw
           )
         end
 
+        UI.message("🤖 Starting automated tests. Note: This feature is in beta.")
+        release_tests = []
+        if present?(test_cases)
+          test_cases.each do |tc|
+            release_tests.push(create_release_test(alpha_client, release.name, device_executions, login_credential, tc))
+          end
+        else
+          release_tests.push(create_release_test(alpha_client, release.name, device_executions, login_credential))
+        end
+        release_tests
+      end
+
+      def self.create_release_test(alpha_client, release_name, device_executions, login_credential, test_case_name = nil)
         release_test =
           Google::Apis::FirebaseappdistributionV1alpha::GoogleFirebaseAppdistroV1alphaReleaseTest.new(
+            device_executions: device_executions,
             login_credential: login_credential,
-            device_executions: device_executions
+            test_case: test_case_name
           )
-        alpha_client.create_project_app_release_test(release.name, release_test)
+        alpha_client.create_project_app_release_test(release_name, release_test)
       rescue Google::Apis::Error => err
-        UI.crash!(err)
+        case err.status_code.to_i
+        when 404
+          UI.user_error!("Test Case #{test_case_name} not found")
```

Comment:
Can we guarantee that any 404 means the test case wasn't found, and not the release/app/project?

Reply:
No. I made the [same comment](https://github.com/FirebasePrivate/firebase-android-buildtools/pull/545#discussion_r1880499200) when Tunde implemented this for Gradle.

But since it's extremely unlikely that the 404 is because the release doesn't exist (after all it was just created/updated due to the upload) and extremely likely that this is caused my a missing test case, I was convinced to add this.

Reply:
Fair!